### DISCUSSION TITLE
[Feature Request] Custom Courses

### DISCUSSION THREAD

**Discussion by DarkBahamut162** [TeamRizu/OutFox#456] [03 - Feature Requests]
### Is your feature request related to a problem?

Somewhat?
I got an idea I would like to implement in my future theme.
But accorrding to the Stepmania GitHub Source Code, I didn't find a way to do so.

This is mostly for Course Mode:
I would like to set a custom Course with a custom length of songs with the ability to make them either random or set.

### Describe the solution you'd like

I would like to be able to do the following:

```
local customCourse = COURSEMAN:CreateCourse();

if selectedCodeMessage then
	customCourse:addSongToList(song/steps,difficulty);
elseif remove CodeMessate then
	customCourse:removeLastSongFromList();
end

GameCommand:SetCourse(customCourse)
```

In short: A way to create customCourses with selected/randomized Songs/Difficulties and set them as currentCourses.

(I'm currently still in the planning phase so it might still not look as fleshed out)

### Describe alternatives you've considered

There are no alternatives!
This would be very unique thing for OutFox, bringing potential people over for something special.

**Comment by Scraticus** [comment]
This is a great idea, we will definitely be taking this on board.
